apps/help-center/utils.js

```diff
@@ -0,0 +1,20 @@
+import { __dangerousOptInToUnstableAPIsOnlyForCoreModules } from '@wordpress/private-apis';
+
+/**
+ * Sometimes Gutenberg doesn't allow you to re-register the module and throws an error.
+ * FIXME: The new version allow it by default, but we might need to ensure that all the site has the new version.
+ * @see https://github.com/Automattic/wp-calypso/pull/79663
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+let unlockRouter;
+try {
+	unlockRouter = __dangerousOptInToUnstableAPIsOnlyForCoreModules(
+		'I acknowledge private features are not for use in themes or plugins and doing so will break in the next version of WordPress.',
+		'@wordpress/router'
+	).unlock;
+} catch ( error ) {
+	// eslint-disable-next-line no-console
+	console.error( 'Error: Unable to get the unlock api. Reason: %s', error );
+}
+
+export { unlockRouter };
```

Comment:
This originally was the code in the `hooks.js` file to get the canvas state from `edit-site`. I needed the `RouterProvider` for the Post views otherwise `location` came back `null`. This probably needs some work.

Reply:
I think we can just return `null` for the post editor as it's expected behavior that the canvas mode is limited to the site editor.

Reply:
[Here](https://github.com/Automattic/wp-calypso/blob/24a8e74579ca0c59cf8f8b1910e131de9c255ec4/apps/wpcom-block-editor/src/wpcom/features/live-preview/hooks/use-location.ts#L8-L15) is what wpcom-block-editor does.